src/asmcnc/job/job_data.py

```diff
@@ -166,6 +166,20 @@ def set_job_filename(self, job_path_and_name):
         else:
             self.job_name = self.filename.split("/")[-1]
 
+
+    def scrape_last_feed_command(self, job_gcode_object, index): 
+
+        try:
+            feedrate_line = next((s for s in reversed(job_gcode_object[:index]) if 'F' in s), None)
+            if feedrate_line:
+                feedrate = re.match('\d+(\.\d+)?$',feedrate_line[feedrate_line.find("F")+1:]).group()
```

Comment:
Is it possible that the feedrate is defined in the middle of the line rather than the end? If this is the case then the $ will stop it from matching since it's not the end of the line. Seems unlikely but worth checking

Reply:
Good spot @DennisKiselev, thank you! Have updated unit tests and removed $ sign :) 

Please check and resolve this conversation if you're happy 😁